File: frontend/src/api/ai.ts
import api from './client';
import type { CaptionResponse, HashtagResponse } from '../types';

export async function generateCaption(
  description: string,
  platforms: string[],
  tone = 'professional',
  keywords?: string[],
) {
  const resp = await api.post<CaptionResponse>('/ai/generate-caption', {
    description,
    platforms,
    tone,
    keywords,
  });
  return resp.data;
}

export async function generateHashtags(caption: string, platform: string, category?: string) {
  const resp = await api.post<HashtagResponse>('/ai/generate-hashtags', {
    caption,
    platform,
    category,
  });
  return resp.data;
}

export async function rewriteCaption(caption: string, sourcePlatform: string, targetPlatform: string) {
  const resp = await api.post<{ rewritten_caption: string }>('/ai/rewrite', {
    caption,
    source_platform: sourcePlatform,
    target_platform: targetPlatform,
  });
  return resp.data.rewritten_caption;
}

export async function generateCalendar(
  startDate: string,
  endDate: string,
  platforms: string[],
  contentThemes?: string[],
  postsPerDay = 1,
) {
  const resp = await api.post('/ai/generate-calendar', {
    start_date: startDate,
    end_date: endDate,
    platforms,
    content_themes: contentThemes,
    posts_per_day: postsPerDay,
  });
  return resp.data;
}

File: backend/app/schemas/ai.py
from pydantic import BaseModel


class CaptionRequest(BaseModel):
    description: str
    platforms: list[str]  # ['instagram', 'twitter', 'facebook', 'tiktok']
    tone: str = "professional"  # professional, casual, funny, inspirational
    keywords: list[str] | None = None


class CaptionResponse(BaseModel):
    captions: dict[str, str]  # platform -> caption
    variations: list[dict[str, str]] | None = None


class HashtagRequest(BaseModel):
    caption: str
    platform: str
    category: str | None = None


class HashtagResponse(BaseModel):
    hashtags: list[str]
    broad: list[str]
    niche: list[str]


class RewriteRequest(BaseModel):
    caption: str
    source_platform: str
    target_platform: str


class RewriteResponse(BaseModel):
    rewritten_caption: str


class CalendarRequest(BaseModel):
    start_date: str  # ISO date
    end_date: str  # ISO date
    platforms: list[str]
    content_themes: list[str] | None = None
    posts_per_day: int = 1


class CalendarSlot(BaseModel):
    date: str
    time: str
    platform: str
    content_type: str
    theme: str
    suggested_caption: str


class CalendarResponse(BaseModel):
    slots: list[CalendarSlot]
